Python/miscellaneous/find_freq_using_bs.py

```diff
@@ -0,0 +1,60 @@
+def get_freq_using_bs(arr: list, N: int, key: int) -> int:
+    """Returns frquency of key in arr 
+
+    This function takes an list and a key and finds 
+    the occurence of that key in the list using 
+    binary search technique.
+
+    Parameters
+    -------------
+    arr : The list in which search is to be performed
+    N :  size of the list
+    key :  The element whose frequency is to be calculated
+
+    Returns
+    -------------
+    count : The frequency of key 
+    """
+    # initializing necessary variables
+    left = 0
+    right = N - 1
+    count = 0
+
+    # Sorting inorder to use binary search
+    # use any stable sorting algorithm
+    arr.sort()
+
+    # Performing binary search with slight modification
+    while left <= right:
+        mid = (left + right) // 2
+
+        if arr[mid] == key:
+            count += 1
+            leftI = mid - 1
+            rightI = mid + 1
+            # counting keys preceding to mid
+            while leftI >= 0 and arr[leftI] == key:
+                count += 1
+                leftI -= 1
+            # counting keys succeding to mid
+            while rightI < N and arr[rightI] == key:
+                count += 1
+                rightI += 1
+            break
+        
+        elif arr[mid] > key:
+            right = mid - 1
+
+        else:
+            left = mid + 1
+
+    return count
+
+
+# driver code
+if __name__ == "__main__":
+    N = int(input())
+    arr = list(map(int, input().split()))
+    key = int(input())
+
+    print(f"Ocurrence of '{key}': {get_freq_using_bs(arr, N, key)}")
```

Comment:
@Kathan-Vakharia  Please replace this line here by `print("Ocurrence of",key,"is :", get_freq_using_bs(arr, N, key))`